From: Paul Blakey <paulb@mellanox.com>
Date: Wed, 14 Dec 2016 19:00:58 +0200
Subject: net/sched: cls_flower: Use masked key when calling HW offloads
Patch-mainline: v4.10-rc1
Git-commit: f93bd17b916959fc20fbb7dc578e1f2657a8c343
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

Zero bits on the mask signify a "don't care" on the corresponding bits
in key. Some HWs require those bits on the key to be zero. Since these
bits are masked anyway, it's okay to provide the masked key to all
drivers.

Fixes: 5b33f48842fa ('net/flower: Introduce hardware offload support')
Signed-off-by: Paul Blakey <paulb@mellanox.com>
Reviewed-by: Roi Dayan <roid@mellanox.com>
Acked-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[ adjusted for the absence of 3036dab670ba ("net/sched: cls_flower:
  Provide a filter to replace/destroy hardware filter functions",
  v4.10-rc1) ]
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 net/sched/cls_flower.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/sched/cls_flower.c
+++ b/net/sched/cls_flower.c
@@ -628,7 +628,7 @@ static int fl_change(struct net *net, st
 	fl_hw_replace_filter(tp,
 			     &head->dissector,
 			     &mask.key,
-			     &fnew->key,
+			     &fnew->mkey,
 			     &fnew->exts,
 			     (unsigned long)fnew,
 			     flags);
